### DISCUSSION TITLE
Collecting user feedback without last_event_id

### DISCUSSION THREAD

**Discussion by nijel** [getsentry/sentry-python#3016] [Q&A] [answered]
`last_event_id()` was deprecated in https://github.com/getsentry/sentry-python/pull/2749 and removed in the 2.0 release.

We use it to display user feedback form on unhandled error (in `handler500`, see https://github.com/WeblateOrg/weblate/blob/5d162af6fd87e1bbb172470d8752a2373049e7c4/weblate/trans/views/error.py#L57-L73). To link user feedback with the exception, the event ID is needed.

Is there now some way to get event ID for events captured inside integrations, and not manually?

The documentation mentions that `capture_event` and other return event ID, but that is not stored anywhere for events captured inside the signal handler:

https://github.com/getsentry/sentry-python/blob/439d418f244f6dd8342c88a40e2196734d98b8fb/sentry_sdk/integrations/django/__init__.py#L512

**Comment by nofalx** [comment]
Facing similar Issue

**Comment by nofalx** [comment]
This worked for me somehow, no duplicate issues

```python
class SentryEventMiddleware:
    def __init__(self, get_response):
        self.get_response = get_response

    def __call__(self, request):
        response = self.get_response(request)
        return response

    def process_exception(self, request, exception):
        event_id = sentry_sdk.capture_exception(exception)
        request.sentry_event_id = event_id


def custom500(request, exception=None):
    """Custom 500"""
    return JsonResponse(
        {
            "message": "Internal Server Error",
            "code": getattr(request, "sentry_event_id", None),
        },
        status=500,
    )
```

**Reply by nijel** [reply]
Thanks for the suggestion, I will probably end up with something like this as well. I was hoping for a built-in solution and not for one everybody will have to copy.

**Reply by nijel** [reply, marked as answer]
Actually, it is planned, see https://github.com/getsentry/sentry-python/issues/3049 and https://github.com/getsentry/sentry-python/pull/3057
